Dockerfile

```
@@ -1,16 +1,28 @@
-FROM debian:bookworm-slim
+FROM python:3.11-slim-trixie AS joinmarket
 
-RUN mkdir -p /jm/clientserver
 WORKDIR /jm/clientserver
+COPY ./pubkeys ./pubkeys
+COPY ./install.sh ./install.sh
+
+RUN apt-get update \
+  && apt-get install -y --no-install-recommends \
+    gnupg \
+    ca-certificates \
+    curl \
+  && ./install.sh --docker-install --deps-only \
+  && apt-get clean \
+  && rm -rf /var/lib/apt/lists/*
 
 COPY . .
 
-RUN apt-get update && apt-get install -y --no-install-recommends gnupg ca-certificates=* curl=* \
-  python3-pip=* python3=* \
-  && pip3 config set global.break-system-packages true \
-  && pip3 install 'wheel>=0.35.1' \
-  && ./install.sh --docker-install \
-  && apt-get purge -y --autoremove python3-pip \
+RUN apt-get update \
+  && ./install.sh --docker-install --no-deps \
   && apt-get clean \
   && rm -rf /var/lib/apt/lists/*
 
+FROM bitcoin/bitcoin:29.2 AS bitcoin
```

```suggestion
ARG BITCOIN_VERSION=29.2
FROM bitcoin/bitcoin:${BITCOIN_VERSION} AS bitcoin
```

The bitcoin image version could be made into a build-arg.